tests/std/tests/P0896R4_ranges_alg_equal/test.cpp

```diff
@@ -63,6 +63,14 @@ constexpr void smoke_test() {
         int const two_ints[] = {0, 1};
         assert(!equal(one_int, two_ints, comp, proj, proj));
     }
+    {
+        // Validate memcmp case
+        int arr1[3]{0, 2, 5};

```

Comment:
these only validate that the memcmp case is correct not that the optimization actually happened. Any ideas on how to test that it actually happened (I don't have any good ones, maybe a struct with stuff written in it's padding?)

Reply:
I don't think there is a way to test it.

Reply:
yeah I can only think of ways that require doing UB